File: Frontend/trackr-administrador/src/app/shared/http/gestion-caja/metodo-pago.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';

import { MetodoPago } from '@models/gestion-caja/metodo-pago';

@Injectable()
export class MetodoPagoService {
  private dataUrl = 'metodoPago/';

  constructor(public http: HttpClient) {}

  consultarTodos(): Observable<MetodoPago[]> {
    return this.http.get<MetodoPago[]>(this.dataUrl + `consultarTodos`);
  }

  consultarPorClave(clave: string): Observable<MetodoPago> {
    return this.http.get<MetodoPago>(this.dataUrl + `consultarPorClave/${clave}`);
  }
}

File: Backend/TrackrAPI/TrackrAPI/Services/GestionCaja/MetodoPagoService.cs
using TrackrAPI.Models;
using TrackrAPI.Repositorys.GestionCaja;
using System.Collections.Generic;

namespace TrackrAPI.Services.GestionCaja
{
    public class MetodoPagoService
    {
        private IMetodoPagoRepository metodoPagoRepository;

        public MetodoPagoService(
            IMetodoPagoRepository metodoPagoRepository
        )
        {
            this.metodoPagoRepository = metodoPagoRepository;
        }

        public IEnumerable<MetodoPago> ConsultarTodos()
        {
            return metodoPagoRepository.ConsultarTodos();
        }

        public MetodoPago ConsultarPorClave(string clave)
        {
            return metodoPagoRepository.ConsultarPorClave(clave);
        }
    }
}
